package.json

```diff
@@ -27,7 +29,14 @@
     "autoprefixer": "^10.4.7",
     "eslint": "8.18.0",
     "eslint-config-next": "12.1.6",
+    "husky": ">=6",
+    "lint-staged": ">=10",
     "postcss": "^8.4.14",
+    "prettier": "^2.7.1",
     "tailwindcss": "^3.1.4"
+  },
+  "lint-staged": {
+    "*.js": "eslint --cache --fix",
```

Comment:
`lint-staged` will pass each `.js` file in a commit as an argument to `eslint --cache --fix`, meaning that matching files will be linted before the commit completes.